hadoop-ozone/client/src/main/java/org/apache/hadoop/ozone/client/rpc/RpcClient.java

```diff
@@ -445,7 +444,18 @@ public void removeBucketAcls(
   @Override
   public Token<OzoneTokenIdentifier> getDelegationToken(Text renewer)
       throws IOException {
-    return ozoneManagerClient.getDelegationToken(renewer);
+
+    Token<OzoneTokenIdentifier> token =
+        ozoneManagerClient.getDelegationToken(renewer);
+    if (token != null) {
+      Text dtService =
+          getOMProxyProvider().getProxy().getDelegationTokenService();
+      token.setService(dtService);
```

Comment:
I think this if for OM HA. Are we taking Om HA in badlands?

Reply:
On a related note it seems leader in OM HA may change which will mean previously obtained token will not be selected by TokenSelector.

Reply:
We will revisit this when full OM HA is implemented. For 0.4 we can assume this will always return the first proxy. We will have to do this because the HA code has been partially implemented in trunk. We don't want to work two different versions that are difficult to merge later.

Reply:
Agree, if those changes are not there in ozone-0.4, the cherry-pick and merge will be messy.